app/models/group.rb

```diff
@@ -47,6 +47,30 @@ class Group < ActiveRecord::Base # rubocop:disable Metrics/ClassLength
   include Contactable
   include ValidatedEmail
   include Globalized
+  include MountedAttr
+
+  self.mounted_attr_categories = {
+    messages: [:letter_address_position,
+               :letter_logo,
+               :text_message_username,
+               :text_message_password,
+               :text_message_provider,
+               :text_message_originator]
+  }
+
+  PROVIDER_VALUES = %w(aspsms).freeze
+  ADDRESS_POSITION_VALUES = %w(left right).freeze
+
+  mounted_attr :text_message_username, :encrypted
+  mounted_attr :text_message_password, :encrypted
+  mounted_attr :text_message_provider, :string, enum: PROVIDER_VALUES,
+                                                default: PROVIDER_VALUES.first
+  mounted_attr :text_message_originator, :string
+
+  # mount_uploader :carrierwave_letter_logo, GroupSetting::LogoUploader, mount_on: 'letter_logo'
+  mounted_attr :letter_logo, :picture
+  mounted_attr :letter_address_position, :string, enum: ADDRESS_POSITION_VALUES,
+                                           default: ADDRESS_POSITION_VALUES.first
```

Comment:
Ich verstehe nicht ganz, warum jetzt diese Attribute `mounted_attrs` sind, und nicht einfach normale Datenbankspalten. Die mounted_attrs sind meiner Ansicht nach nur gedacht für Attribute, die nur für einzelne Gruppentypen vorhanden sein sollten, also z.B. ein Attribut das auf Kantonalverbänden vorhanden sein soll, aber auf dem Dachverband nicht etc.

Reply:
Ja hab ich auch gedacht. Anhand des Issuebeschriebs Zwischenzeitlich dachte ich noch dass diese Attribute (oder vielleicht nur ein Teil?) nur auf `self.layer = true` Gruppentypen vorhanden seien sollen..
So wie sie jetzt sind

Reply:
Aha! Ja stimmt. Auch für https://github.com/hitobito/hitobito_sww/issues/144 und https://github.com/hitobito/hitobito_sww/issues/146 brauchen wir einen Mechanismus, um mounted_attrs auf allen `self.layer == true` Gruppen einzuführen. Mein Vorschlag wäre, in Zukunft solche Layer-only Attrs in einem `if self.layer?` Block in group.rb einzufügen, und das können wir bei der Implementation der obigen Features machen. Alle anderen, globalen Attrs (also alle die du bisher als `mounted_attr`s erfasst hast) sollten wir aus meiner Sicht noch in diesem PR in normale Attribute ändern.

Reply:
Ja leider funktioniert das nicht rein per Block im `group.rb`
Denn: Die `self.layer` wird erst nach dem Laden der `group.rb` gesetzt (Weil zur Vererbung wird erst die Superklasse geladen)
Mittels initializer funktioniert das ganze aber problemfrei.

Aber sonst passts dann pack ich das noch in normale Attribute, die `migrate_settings` Migration muss ich demfall auch noch anpassen :)